API/services/MealsServices.js

```diff
@@ -0,0 +1,21 @@
+import mealData from "../utils/mealData";
+import Meal from "../model/mealModels";
+
+const MealService = {
+  fetchAllMeals() {
+    const validMeals = mealData.meals.map(meal => {
```

Comment:
The `fetchAllMeals` method should just fetch all the meals. It shouldn't have to iterate over the arrays of meals. Modify this logic to only return the meals that you have in your array. Also, move the modified logic to the fetchMeals method in the MealController. That's where you are supposed to write the logic for fetching all meals.